### ISSUE TITLE
Keyerror when runing GIS with url, username and password

### ISSUE THREAD

**Issue by wondie** [Esri/arcgis-python-api#440]
Below is the error message I got when running:
gis = GIS("https://domain/arcgis/admin", username="name", password="paa")

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python37\lib\site-packages\arcgis\gis\__init__.py", line 452, in __in
it__
    self._is_agol = self._portal.is_arcgisonline
  File "C:\Python37\lib\site-packages\arcgis\_impl\portalpy.py", line 1308, in i
s_arcgisonline
    return self._properties['portalName'] == 'ArcGIS Online' \
KeyError: 'portalName'

It looks like there is a missing parameter somewhere.

**Comment by achapkowski** [comment]
There should be no /admin to connect to a portal.

```
gis = GIS(url="https://mysite.com/portal", username='name', password='paa')
admin = gis.admin
```

**Comment by wondie** [comment]
We don't have portal. We only have arcgis server. http://climatewizard.ciat.cgiar.org/arcgis/admin/login

**Comment by avstjohn** [comment]
Same exact issue here. Trying to script against the Server without using Portal.

**Comment by fallomj** [comment]
We also have the same issue when logging into ArcGIS Server without using Portal.

**Comment by CheckMinus** [comment]
Our temporary fix was to comment out lines 451-453
…\Programs\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\Lib\site-packages\arcgis\gis_init_.py. It appears the _product_version is only called here and never used again in the code.

**Comment by avstjohn** [comment]
And then you were able to create your GIS instance, and do all the fun stuff one can do with the API? Did you have to re-compile the library?

**Comment by pcsswamin** [comment]
Yes. you can connect to your ArcGIS Server via arcgis-python-api. Try the following snippet

```
from arcgis.gis.server import Server

def get_gis_server(server_url):
    gis_server = Server(url=server_url,
                        username='siteadmin',
                        password='pwd123')
    return gis_server

server_url = "https://server.domain.com:6443/arcgis"

gis_server = get_gis_server(server_url=server_url)
```

**Comment by jhevenor** [comment]
Same boat here. Standalone server (no portal).  I can create the Server object and can read its url, and svr.content.url,  but most of the properties described in the documentation are missing (machines, users, etc). 

This bug is months old with no comment, missing two releases.  Has anyone got this working? Have I missed something on my end (running v1.8.0)?

**Comment by CheckMinus** [comment]
The issue got worse at 1.5.0+ 
Below is a link to my fix for 1.6.0+
I have not tested on 1.8.0 but the code base appears at first glance to unchanged.
 
https://community.esri.com/message/896496-re-error-connecting-to-arcgis-server-with-arcgis-python-api?commentID=896496#comment-896496

**Comment by jyaistMap** [comment]
hi @jhevenor @CheckMinus 
I was able to connect to a stand-alone server (with web adaptor installed) using the following. Does this work for you?

```python
server2_base_url = "https://<machine_url>"

server2 = Server(url=f"{server2_base_url}/<web_adaptor_name>/admin",
                token_url=f"{server2_base_url}/<web_adaptor_name>/tokens/generateToken",
                username="<admin_user_name>",
                password="<password>",
                verify_cert=False)
```
